Runtime/SaveComponent.cs

```diff
@@ -210,6 +211,13 @@ public override void OnInspectorGUI()
 			SaveComponent sl = target as SaveComponent;
 			if (sl == null) return;
 
+			if (string.IsNullOrEmpty(sl.id)) sl.GenerateID();
+
+			if (GUILayout.Button("Regenerate"))
+			{
+				sl.GenerateID();
```

Comment:
We should think about automating the checks for naming collisions...

Reply:
no